src/plugins/discover/public/application/main/services/discover_data_state_container.ts

```diff
@@ -221,6 +226,34 @@ export function getDataStateContainer({
         getInternalState,
         savedSearch: getSavedSearch(),
         useNewFieldsApi: !uiSettings.get(SEARCH_FIELDS_FROM_SOURCE),
+      };
+
+      abortController?.abort();
+      abortControllerFetchMore?.abort();
+
+      if (options.fetchMore) {
+        abortControllerFetchMore = new AbortController();
+
+        const fetchMoreStartTime = window.performance.now();
+        await fetchMoreDocuments(dataSubjects, {
+          abortController: abortControllerFetchMore,
+          ...commonFetchDeps,
+        });
+        const fetchMoreDuration = window.performance.now() - fetchMoreStartTime;
+        reportPerformanceMetricEvent(services.analytics, {
+          eventName: 'discoverFetchMore',
+          duration: fetchMoreDuration,
+        });
```

Comment:
@davismcphee 

> It could be good to add a discoverFetchMore performance metric with reportPerformanceMetricEvent even if we don't add a performance journey for it until a later PR.

Added